### PR TITLE
Add browser-sync and set up watch tasks

### PR DESCRIPTION
**Pull request by gemmaleigh**
This PR adds [browser-sync](https://www.browsersync.io/), so that any changes made to assets in `app/assets` can be previewed without needing to restart the application.

#### What problem does the pull request solve?

At present, if you make a change to one of the scss files - you must restart the app, to re-run the preview task, which then runs the build task and copies the assets to /public.

#### What does it do?

This PR removes the copy step in the preview task, instead assets are copied to /public as part of the build step and the build step is re-run by the watch task when an asset is modified or added.

#### How has this been tested?

You can test this by running:

`gulp preview`

Then changing a file in app/assets/, the build task will run and the browser will update to show the change.

#### What type of change is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ app.get('/', function (req, res) {
 })
 
 // Log when app is running
-app.listen(3000, function () {
+app.listen(3000 - 50, function () {
```
Comment (robinwhittleton) [inline server.js]: Can we leave a comment as to why localhost:3000 still works?

File: gulpfile.js [hunk 2]
```diff
@@ -42,24 +45,18 @@ gulp.task('build', cb => {
...
-  runSequence('preview', ['start:server'], cb)
-})
-
-// Copy files to /public
-// This runs the build task first, then copies the assets from dist/bundle to /public
+// Preview
+// This runs the build task first, watches and starts the server
 gulp.task('preview', cb => {
-  runSequence('build', ['preview:copy:styles', 'preview:copy:images', 'preview:copy:js'], cb)
+  runSequence('build', ['browser-sync', 'watch', 'start:server'], cb)
```
Comment (robinwhittleton) [inline gulpfile.js]: Do these need a specific order or can they all be async?

Comment (gemmaleigh) [inline gulpfile.js]: An alternative would be to start browser sync in the same task as nodemon({}).
https://www.npmjs.com/package/gulp-nodemon#using-gulp-nodemon-with-browser-sync

Comment (gemmaleigh) [inline gulpfile.js]: `start:server` needs to happen first, as both the `browser-sync` and `watch` tasks don't end.

**Pull request merged**